src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -1,32 +1,29 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class FigureSupplier {
-    private static final int FIGURE_COUNT = 5;
-    private ColorSupplier colorSupplier = new ColorSupplier();
+    private static final int FIGURE_LIMIT = 5;
+    private Random random = new Random();
 
     public Figure getRandomFigure() {
-        int random = (int) Math.ceil(Math.random() * FIGURE_COUNT);
+        int randomCase = random.nextInt(FIGURE_LIMIT) + 1;
         Figure figure = null;
-        switch (random) {
+        switch (randomCase) {
             case 1:
-                figure = new Circle("circle",
-                            colorSupplier.getRandomColor());
+                figure = new Circle();
                 break;
             case 2:
-                figure = new IsoscelesTrapezoid("isosceles trapezoid",
-                            colorSupplier.getRandomColor());
+                figure = new IsoscelesTrapezoid();
```

Comment:
**IN ADVANCE** don't pass expressions in constructor arguments. let's create separate variables and pass them. and to keep code clean do all of these in a private method.